### PR TITLE
Calendar contact infrastructure

### PR DESCRIPTION
**Pull request by rycee**
Preparing #654 for merge to master.

### CODE REVIEW

**Comment by teto** [comment]
super excited about this one. I think once https://github.com/rycee/home-manager/pull/654#pullrequestreview-262097193 is adressed, we could merge this.

File: modules/accounts/contacts.nix [hunk 1]
```diff
@@ -55,23 +56,21 @@ let
...
-        '';
-      };
+      # userNameCommand = mkOption {
+      #   type = types.nullOr (types.listOf types.str);
+      #   default = null;
+      #   example = [ "~/get-username.sh" ];
+      #   description = ''
+      #     A command that prints the user name to standard output.
+      #   '';
+      # };
```
Comment (jorsn) [inline modules/accounts/contacts.nix]: The same as for [modules/accounts/calendar.nix line 59ff](https://github.com/rycee/home-manager/blob/8450842e4957915243decef882f3cece2b54463d/modules/accounts/calendar.nix#L59) (see above) applies here.

**Comment by jorsn** [comment]
What has to happen for this to be merged into master?
I would help with rebasing, maybe also with refining. I'll do this anyway, because I'm maintaining this as my local fork.

Additionally, I have a module (see [branch](https://github.com/jorsn/home-manager/tree/calendar-contact-infra.khard)) for [khard](https://github.com/scheibler/khard), which I would prepare and contribute if this PR would probably be merged soon.

**Comment by DamienCassou** [comment]
I'm interested as well. What needs to be done?

**Comment by jorsn** [comment]
@rycee Can we merge this when I have added khard and maybe pimutils/todoman (and the additions are fine? If this is the case, I will probably prepare my additions for merge quality rather sooner than later. If not, I'll probably outsource the modules and put them in my NUR or something like that, because rebasing a fork which always conflicts in `news.nix` is unnecessary noise to me.

(BTW: I think this conflict does not occur in e.g. darcs, pijul, see <http://r6.ca/blog/20110416T204742Z.html>)

**Pull request closed without merging**